src/main/kotlin/org/rust/ide/hints/RsVcsCodeVisionContext.kt

```diff
@@ -0,0 +1,26 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.hints
+
+import com.intellij.codeInsight.hints.VcsCodeVisionLanguageContext
+import com.intellij.openapi.editor.Editor
+import com.intellij.psi.PsiElement
+import org.rust.lang.core.psi.RsFunction
+import org.rust.lang.core.psi.ext.RsStructOrEnumItemElement
+import java.awt.event.MouseEvent
+
+@Suppress("UnstableApiUsage")
+class RsVcsCodeVisionContext: VcsCodeVisionLanguageContext {
+    override fun handleClick(mouseEvent: MouseEvent, editor: Editor, element: PsiElement) {}
+
+    override fun isAccepted(element: PsiElement): Boolean {
+        return when (element) {
+            is RsFunction -> true
```

Comment:
Possibly this could be expanded to all qualified named elements? Constants/statics, macros, enum variants, struct fields, type/trait aliases?

Reply:
Yes, I think we can add some more items here, but not too many. I would say that we could add `RsTraitOrImpl` and `RsMacroDefinitionBase` here as potentially large and meaningful code pieces.

Reply:
Ok, I added traits, impls, macros and constants/statics.

Reply:
Thanks! Regarding the constants, maybe reduce their visibility to `RsMod` and `RsMembers` only? Otherwise, code vision is shown for constants declared in function bodies as well.

Also, I forgot to mention `RsModItem` as a good candidate here.

Reply:
I actually wasn't sure about `mod {}`, but it probably doesn't hurt. Good point with the constants, I changed it.

Reply:
Do we really want to add hint for const items?
As I can see, neither Java nor Kotlin don't provide hints for them.
Also, it looks quite inconsistent that constants have author hints but type aliases don't
Maybe it's better to drop hints for constants?

Reply:
It sounds reasonable to me (this is an important constant, who was the last one to change it?). But consistency is maybe more important. I don't think that type aliases are as important as constants. I'll remove constants.